### PR TITLE
Fix tenant label for custom tenant when both Global and Private tenan…

### LINKED ISSUE #1248: [BUG] Dashboard login problem - global_tenant getting displayed in custom tenants if Global tenant is disabled
**What is the bug?**
Dashboard login has a problem that if we give access to global_tenant for some user/role, then global_tenant is getting displayed in custom tenants if Global tenant is disabled.

**How can one reproduce the bug?**
Pre-requisites:

1. We have a custom tenant "cadm"
2. We have disabled "Global" and "Private" tenants

1. Go to Dashboard
2. Click on Single Sign-on
3. See that user is logged in to global_tenant by default:
global-tenant-issue-1

4. Once the user selects the custom tenant "cadm", then the global_tenant automatically gets removed from custom tenant drop down:
global-tenant-issue-2

**What is the expected behavior?**
We should not see global_tenant in custom tenants drop down if Global tenant is disabled from multi tenancy configuration.

**What is your host/environment?**
 - Plugins

**Do you have any screenshots?**

**Do you have any additional context?**
Opensearch version 2.4.0

### PR DESCRIPTION
**Pull request by RyanL1997**
Signed-off-by: Ryan Liang <jiallian@amazon.com>
### Description
Fix tenant label for custom tenant when both Global and Private tenants are disabled
### Category
Bug fix
### Issues Resolved
* Resolved #1248 
### Check List
- [ ] ~New functionality includes testing~
- [ ] ~New functionality has been documented~
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/multitenancy/tenant_resolver.ts [hunk 1]
```diff
@@ -143,7 +143,18 @@ function resolve(
...
+   * fall back to the first tenant in the available tenants
+   * Under the condition of enabling multitenancy, if the user has disabled both 'Global' and 'Private' tenants:
+   * it will remove the default global tenant key for custom tenant.
+   */
+  if (
+    Object.keys(availableTenantsClone).length > 1 &&
+    availableTenantsClone.hasOwnProperty(globalTenantName)
+  ) {
+    delete availableTenantsClone[globalTenantName];
+    return findKey(availableTenantsClone, () => true);
```
Comment (cwperks) [inline server/multitenancy/tenant_resolver.ts:156]: The return here isn't necessary because it will hit the same return statement when it exits the block.

**Review by cwperks — CHANGES_REQUESTED** [review]
Thank you @RyanL1997! The change looks good to me, but I'd like to see a test to verify the change.

File: server/multitenancy/tenant_resolver.ts (hunk 1 above)
Comment (RyanL1997) [reply server/multitenancy/tenant_resolver.ts:156]: Nice catch @cwperks! I will update this with the test case.

File: server/multitenancy/test/tenant_resolver.test.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      availableTenants: { global_tenant: true, admin_tenant: true, test_tenant: true, admin: true },
+      globalTenantEnabled: false,
+      privateTenantEnabled: false,
+    };
+
+    const adminResult = resolveWithConfig(adminConfig);
+    expect(adminResult).toEqual('admin_tenant');
+  });
+
+  it('Resolve tenants list for non-admin user', () => {
```
Comment (cwperks) [inline server/multitenancy/test/tenant_resolver.test.ts:44]: Do these tests have anything to do with whether the user is admin or non-admin?

What do you think about:

1. Resolve tenant with custom tenants, Global and Private disabled
2. Resolve tenant without custom tenants, Global and Private disabled

File: server/multitenancy/tenant_resolver.ts [hunk 3]
```diff
@@ -143,7 +143,17 @@ function resolve(
...
+  /**
+   * Fall back to the first tenant in the available tenants
+   * Under the condition of enabling multitenancy, if the user has disabled both 'Global' and 'Private' tenants:
+   * it will remove the default global tenant key for custom tenant.
+   */
+  if (
+    Object.keys(availableTenantsClone).length > 1 &&
+    availableTenantsClone.hasOwnProperty(globalTenantName)
+  ) {
+    delete availableTenantsClone[globalTenantName];
```
Comment (cwperks) [inline server/multitenancy/tenant_resolver.ts:155]: This will work, but it's a bit confusing why `global_tenant` is in this data structure in the first place. I went over to the security plugin to find where its coming from and it looks like its being added here: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java#L1135-L1140

I'm finding that block hard to follow and not sure what the reason for it is.

This PR looks good to me overall. I will approve when the test descriptions are updated.

File: server/multitenancy/test/tenant_resolver.test.ts (hunk 2 above)
Comment (RyanL1997) [reply server/multitenancy/test/tenant_resolver.test.ts:44]: Yes, that makes sense, I was using this because as the admin user I think you will have the `admin_tenant` as one of the reserved custom tenants, but you can always change that.

File: server/multitenancy/tenant_resolver.ts (hunk 3 above)
Comment (RyanL1997) [reply server/multitenancy/tenant_resolver.ts:155]: I think this `global_tenant` is basically functioning as the 'Global' tenant. Just for case that user doesn't have any custom tenants with multi-tenancy enabled but disable both 'Global' and 'Private' tenants.

Referencing to the information from @cliu123, I also left a comment in the original issue: https://github.com/opensearch-project/security-dashboards-plugin/issues/1248#issuecomment-1359959042

File: server/multitenancy/test/tenant_resolver.test.ts (hunk 2 above)
Comment (RyanL1997) [reply server/multitenancy/test/tenant_resolver.test.ts]: Done!

**Comment by cliu123** [comment]
LGTM！

File: server/multitenancy/tenant_resolver.ts (hunk 3 above)
Comment (cliu123) [reply server/multitenancy/tenant_resolver.ts:155]: @cwperks Yes, `global_tenant` comes from there.
@RyanL1997 You're right. `global_tenant` is global tenant.

**Review by cliu123 — APPROVED** [review]

**Review by cwperks — APPROVED** [review]
